### ISSUE TITLE
Add place to enter max hp increases without having to go into "edit" mode (same as temp hp works)

### ISSUE THREAD

**Issue by oakes123** [foundryvtt/dnd5e#3084]
Add place to enter max hp increases without having to go into "edit" mode (same as temp hp works).  This is a pretty common use case that would be nice to be able to do while in game mode.

**Comment by JPMeehan** [comment]
Is this all that common? It's basically just the _Aid_ spell, which has an Active Effect now you can apply so you don't need the UI spot.

**Comment by BrotherSharper** [comment]
There are plenty of monsters that reduce max HP, starting with the most common being vampires.
I found it clunky as well having to enter into edit mode to change max HP

**Comment by Vengrath97** [comment]
Yes, this is something that really often comes up, especially Max HP reduction, and is really important to have. Earlier versions had it I think, or was it just 5eTidySheet?
Anyway, with the new, enhanced sheet, such a field would be greatly usefull!

**Comment by oakes123** [comment]
I think some tables don't really use the automated spell stuff in active effects and prefer to just manually edit effects (as well as the monster cases mentioned above).

**Comment by Fyorl** [comment]
I'm not convinced this is common enough to warrant a dedicated spot in play mode on the sheet. Temporary Hit Points is a codified rule by comparison, whereas 'life drain' effects only exist as ad-hoc rules text in a tiny handful of creature stat blocks.

**Comment by BrotherSharper** [comment]
I don't believe this conversation should go to whether something is codified, but rather if it can occur using the most basic form of PHB + DMG + MM combination when playing.

I want to propose the following perspectives.
- This used to exist in the old sheet and made an already existing feature harder to access.
- The sheet already supports extraordinary effects from the Star icon, it would still be easier to access from there.
- Even if it does not have its own UI, a shortcut, i.e. right clicking temp HP could bring up the menu to reduce max HP.

Would it help if I went and fetched statistical data per CR and amount of monsters in the MM that do have this ability, VS the actual occurrence of another UI feature?
i.e. what would we need to provide to objectively prove that this warrants a space in the UI?

I run D&D 5 times a week (90% wotc material) and this reality does not align with my own experiences.

**Comment by oakes123** [comment]
My experience is also that this comes up relatively often (why I wrote the request - we played a game and could not intuitively figure out how to do it, and even when we know how to do it is it clunky).  

I don't think counting the monsters that have this ability gives it enough weight anyway.  I can't think of too many tables that don't run into spectres, wraiths, vampires at some point - these are iconic undead monsters.

Aid and Heroes Feast are very commonly used spells and both have this effect as well.

**Comment by Vengrath97** [comment]
There is also the 4th level exhaustion drawback, that can be influenced by MaxHP gain/loss

**Comment by Fyorl** [comment]
I think Aid and Heroes' Feast are bad examples since those spells have very long durations. They are not examples that involve frequently having to change the maximum HP of a character.

I acknowledge that for some tables that are frequently fighting some specific monsters, that they may also have to adjust maximum hit points frequently, but that is not a convincing argument. There are many other mechanics that you may encounter disproportionately, depending on any number of factors. For example, if you happen to be fighting lots of Bearded Devils or your DM really likes their 'wounded' mechanic, you would disproportionately find yourself dealing with that mechanic, for which there is also no support on the core sheet for (the tracking of stacking wounds). A sorcerer that is creating additional spell slots over their maximum with Flexible Casting might find themselves using edit mode a lot to override their spell slot maximum temporarily, but that is not a good argument for spell slot maximums being editable in play mode.

**Comment by JPMeehan** [comment]
Is there a good way to incorporate the max HP reduction effects onto the relevant attacks? That seems like a better solution.

**Comment by oakes123** [comment]
tidy5e did this in a cave with a box of scraps!

This is the wrong place to save space because almost no space would be consumed by this.

The current default sheet has plenty of places it could save far more space if that is the intent.

Having it tacked onto automated damage application is a bad solution for anyone that isn't using that.

**Comment by BrotherSharper** [comment]
I don't believe wounded and max spell slots should be brought up here.
Stacking status effects and easily overridable max spell slots are valid suggestions in themselves. Another QoL improvement not being implemented is not a valid argument against the current QoL improvement we are asking for here. (Not even an improvement but rather a "bring back the previous thing" kind of request)

Frequency is also not a good argument, as there are more monsters that reduce max HP than monsters that petrify a character. If we go down that route, we should get rid of petrified status since it can be dealt with by combining other existing status effects.

The main point here IMO is that It's an existing feature that was rendered hard to access, and thus DMs/players that used to rely to it are having a poorer user experience with the new sheet. 

Please don't take this as a "must fix now" kind of adversarial thing, but please add it to the backlog of future QoL that the sheet could receive. Unless there is some kind of demand from WotC to keep it in line with D&D Beyond's UI (which also lacks the temp max HP reduction), I don't see why we are even having these arguments about it.

**Comment by Fyorl** [comment]
> I don't believe wounded and max spell slots should be brought up here.
Stacking status effects and easily overridable max spell slots are valid suggestions in themselves. Another QoL improvement not being implemented is not a valid argument against the current QoL improvement we are asking for here.

This is the reason we are having a discussion about it, I think. Because if making anything editable in play mode is framed as a 'QoL improvement' then we should just make everything editable in play mode, and there would be no need for a toggle at all. We have to draw the line somewhere, we cannot design a sheet where every part of the Actor data model is editable with a single click.

That line can move; I'm not claiming the current design is perfect, I'm only saying that, for me, I'm not currently convinced that `tempmax` is something that needs editing in play mode.

**Comment by BrotherSharper** [comment]
It is to be framed as a QoL since, in fact, this entire thread appeared because somebody thought it hard to navigate the existing UI. I wouldn't post here this much if I didn't think this was a QoL.

I value your subjective view; in that case, what do you think the best way is to move this forward?

I am grateful for all the effort put into the new sheet, but due to its suddenness, it hasn't had enough time to sift through discussions like this one.

How do we move forward to have the line redrawn?
Community poll, petition, live data etc.

**Comment by Vengrath97** [comment]
Then maybe at the very least include it in the edit mode as an editable field, between current HP and TempHP ?

**Comment by oakes123** [comment]
To get to add any temporary max hp you have to first enter edit mode then you have to hit the cogwheel for hit points, and then you're presented with a dialog that has two things that are static during almost all kinds of play (bonus per level and overall bonus) and one thing that is absolutely not static during many kinds go play.

- Best solution to me is the old solution and the one that tidy5e does - there is a field right next to temp hp in play mode that you can adjust as needed in play.
- Next best solution would be turning on "edit mode" and an adjustment field is presented rather than being buried in a dialog with two other things it is not similar to.

If there is some kind of design philosophy behind the "edit mode" vs "play mode" it would be great to see it articulated.   What does edit mode aspire to accomplish?  What does play mode aspire to cover?  When there are conflicts is there a logic that can be followed?  My perception based on this discussion is that each item brought up is debated on its own individual merits without any guiding principles.

**Comment by JPMeehan** [comment]
I think there's a reasonable ask here that the one click to go to Edit Mode is reasonable, but max HP adjustments are common enough that you shouldn't need to open a *deeper* dialog.

**Comment by oakes123** [comment]
There is another issue now that I look more closely and compare the behavior of the three settings on that dialog.  This may be a bug but it is connected to the design choices here.

While I can enter a "temp max hp" on the dialog and save it, it doesn't seem to change anything.  My max hp hasn't changed, or if it has it isn't being displayed on the sheet as different.

The other two fields on the dialog seem to directly change the sheet when I save changes.

**Comment by oakes123** [comment]
Actually counterintuitively for the "temp max hp" to work I have to save and then hit save again.  The second time I do this the value changes and the dialog closes itself.  If I save once, no change even if I close the dialog.

**Comment by krbz999** [comment]
The dialog for adjusting HP re-renders when an input field is changed (the same occurs in the AC config). Your first click on the Save button is likely absorbed by this. Changes are only made when the config actually closes as part of hitting the Save button.

**Comment by oakes123** [comment]
> The dialog for adjusting HP re-renders when an input field is changed (the same occurs in the AC config). Your first click on the Save button is likely absorbed by this. Changes are only made when the config actually closes as part of hitting the Save button.

I thought it might be a rendering issue too but it seems to be more than that.  Try this:

- Open the dialog
- Enter a value for temp max hp
- Hit save once
- Notice that nothing changed
- Close the dialog
- Notices that still nothing changed
- Open the dialog
- Notice that the value you entered is gone
- Enter another value
- Hit save twice
- Notice that max hp has changed and the dialog closed itself
- Open the dialog
- See that the value you entered is there
